plugins/jenkins/app/models/samson/jenkins.rb

```diff
@@ -30,17 +56,129 @@ def initialize(job_name, deploy)
       @deploy = deploy
     end
 
+    def jenkins_job_config
+      conf = Rails.cache.fetch(job_name + "_conf", expires_in: 7.days, race_condition_ttl: 5.minute) do
+        conf = client.job.get_config(job_name)
+      end
+      Nokogiri::XML(conf)
+    end
+
+    def configure_job_config
+      conf = jenkins_job_config
+      post_config = false
+      exp_build_params = JENKINS_BUILD_PARAMETERS.keys
+      present_build_params = get_build_params(conf).map(&:to_sym)
+      missing_build_params = exp_build_params.to_set - present_build_params.to_set
+      unless missing_build_params.empty?
+        add_build_parameters(conf, missing_build_params)
+        post_config = true
+      end
+      unless check_description(conf)
+        add_job_description(conf)
+        post_config = true
+      end
+      if post_config
+        post_job_config(conf)
+      end
+    end
+
+    def get_build_params(conf)
+      params = conf.xpath("//parameterDefinitions").first
+      params_array = []
+      if params
+        params.children.each do |param|
+          if param.name == "hudson.model.StringParameterDefinition"
+            param.children.each do |value|
+              if value.name == "name" && value.content.starts_with?(JENKINS_BUILD_PARAMETRS_PREFIX)
+                params_array.append(value.content.split(JENKINS_BUILD_PARAMETRS_PREFIX)[1])
+              end
+            end
+          end
+        end
+      end
+      params_array
+    end
+
+    def job_names_from_desc(content, idx)
+      job_names = []
+      while idx < content.size && content[idx] != JENKINS_JOB_DESC[1][-1]
+        job_names.append(content[idx]) if content[idx].starts_with?('*')
+        idx += 1
+      end
+      [idx, job_names]
```

Comment:
maybe:
```Ruby
job_names = content[ids..content.size].select { |c| c != JENKINS_JOB_DESC[1][-1] && c.start_with?('*') }
```

... otherwise please provide some docs on how this operates ... looks super complicated :D